scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3138,7 +3138,14 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     def aliasType() = Defn.Type(mods, name, tparams, typ())
     def abstractType() = Decl.Type(mods, name, tparams, typeBounds())
     token match {
-      case Equals() => next(); aliasType()
+      case Equals() =>
+        next()
+        if (token.is[KwImplicit] && dialect.allowImplicitFunctions) {
+          next()
+          Defn.Type(mods ++ immutable.Seq(Mod.Implicit()), name, tparams, typ())
+        } else {
+          aliasType()
+        }
```

Comment:
Looking at [the tests](https://github.com/lampepfl/dotty/pull/1775/files#diff-21900f439c1e991ad21e8c9be8c80fe8R6) in the PR linked from https://www.scala-lang.org/blog/2016/12/07/implicit-function-types.html there's also 

```
val i1 = (implicit (s: String) => s.length > 2)
val x: implicit String => Boolean = { implicit (s: String) => s.length > 2 }
```